lib/modules/opengraph_builder.rb

```diff
@@ -0,0 +1,60 @@
+# frozen_string_literal: true
+
+# This class is meant to be used as a wrapper for
+# Opengraph meta-tags within HTML markup.
+#
+class OpengraphBuilder
+  attr_accessor :data
+
+  # Create an instance e.g. og = OpengraphBuilder.new
+
+  # The initializer accepts a hash with the starting structure for example...
+  #
+  # og = OpengraphBuilder.new(
+  #   { og: { title: 'my title' }, twitter: { card: 'my_card' } }
+  # )
+  def initialize(content = {})
+    @data = Hash.new({}).merge(content.stringify_keys)
+  end
+
+  # Use og.content with a prefix and hash to store, for example...
+  #
+  # og 'twitter', card: 'content',
+  #               title: 'Example title!',
+  #               description: 'Example description',
+  #               site: '@example',
+  #               image: 'https://placehold.it/100',
+  #               creator: '@example' %>
+  #
+  # To load this content into a layout view, you would then simply call...
+  #
+  # <%== og.content %> to get all of the content in the store, or...
+  #
+  # <%== og.content('twitter') %> to get only the content for a specific prefix.
+  #
+  def content(prefix = nil, **options)
+    if options.empty?
+      return meta_tags(@data[prefix], prefix) if prefix
+
+      arr = []
+      @data.each { |p, data| arr << meta_tags(data, p) }
+      arr.join("\n")
+    else
+      prefix = prefix ? prefix.to_s : 'og'
+      @data.store(prefix, @data[prefix].merge(options))
+    end
+  end
+
+  private
+
+  def meta_tags(data, prefix)
+    arr = []
+    data.each do |key, value|
+      arr << format(
+        '<meta property="%<prefix>s:%<key>s" content="%<value>s">',
+        { prefix: prefix, key: key, value: CGI.escapeHTML(value.to_s) }
```

Comment:
I noticed that the `CGI.escapeHTML` is causing this
![Screenshot from 2020-07-24 10-53-24](https://user-images.githubusercontent.com/40596660/88380177-f60ea000-cd9b-11ea-93c9-075eca6cb90f.png)
Not sure you really need that method, maybe just `value.to_s` is enough, but I may be missing something here

Reply:
The issue was that if there are no results from locales, it could give you a string that would make the HTML very unhappy. I wonder if there is a simpler escaping method I could use in this case?

Reply:
On second inspection, it does seem to be sufficient actually, with just `.to_s` with a broken locale:

`<meta property="og:title" content="translation missing: en.opengraph.defaults.titleadsa">`